src/jaxsim/parsers/rod/utils.py

```diff
@@ -206,3 +217,62 @@ def fibonacci_sphere(samples: int) -> npt.NDArray:
     return descriptions.SphereCollision(
         collidable_points=collidable_points, center=center_wrt_link
     )
+
+
+def create_mesh_collision(
+    collision: rod.Collision,
+    link_description: descriptions.LinkDescription,
+    mesh_uri: str,
+    method: MeshMappingMethods = MeshMappingMethods.VertexExtraction,
+    nsamples: int = 1000,
+) -> descriptions.MeshCollision:
+    file_obj = resolve_local_uri(uri=collision.geometry.mesh.uri)
+    try:
+        _file_type = str(file_obj).split(".")[-1]
+    except ValueError as e:
+        raise e(f"Failed to get file type from {file_obj}")
+    logging.debug(
+        f"===> Loading mesh {collision.geometry.mesh.uri} with scale {collision.geometry.mesh.scale}, file type '{_file_type}'"
+    )
+    mesh = trimesh.load(
+        file_obj=open(file_obj, "rb"),
+        file_type=_file_type,
+    )
```

Comment:
Is the file correctly close afterwards? I'd suggest to use a context manager guarding these lines.

Reply:
I see from https://trimesh.org/trimesh.html#trimesh.load that `trimesh.load` accepts a `pathlib.Path`. Can you use it instead of a file object? Furthermore, since the URI can only point to mesh files, what about using `trimesh.load_mesh`?

Reply:
I suspect that using `pathlib.Path` objects as suggested, trimesh might be smart enough to automatically detect the extension without the need to explicitly extract the suffix.